translator/GCP_cloud_function/cloud_function/main.py

```diff
@@ -1,35 +1,53 @@
-# {'bucket': 'icone_files', 'contentType': 'image/jpeg', 'crc32c': 'ZO4N6A==', 'etag': 'CNu+up7tnu4CEAE=', 
-# 'generation': '1610745138552667', 'id': 'icone_files/61862931_1039122999618882_7010636350885986304_o_1039122996285549.JPEG/1610745138552667', 
-# 'kind': 'storage#object', 'md5Hash': 'yOlwCtFi0L/3GPe23rYGbw==', 
-# 'mediaLink': 'https://www.googleapis.com/download/storage/v1/b/icone_files/o/61862931_1039122999618882_7010636350885986304_o_1039122996285549.JPEG?generation=1610745138552667&alt=media', 
-# 'metageneration': '1', 'name': '61862931_1039122999618882_7010636350885986304_o_1039122996285549.JPEG',
-#  'selfLink': 'https://www.googleapis.com/storage/v1/b/icone_files/o/61862931_1039122999618882_7010636350885986304_o_1039122996285549.JPEG',
-#   'size': '447421', 'storageClass': 'STANDARD', 'timeCreated': '2021-01-15T21:12:18.552Z', 'timeStorageClassUpdated': '2021-01-15T21:12:18.552Z', 
-#   'updated': '2021-01-15T21:12:18.552Z'}
-
 import icone_translator
 from gcloud import storage
 import json
 import xmltodict
+import shutil
+import urllib.request as request
+from contextlib import closing
+from google.cloud import pubsub_v1
+
+
+server = 'iconetraffic.com'
+port = '42663'
+user = 'cdot'
+password = 'icone_cdot'
+filepath = 'incidents.xml'
+
+
+ftpString = 'ftp://{0}:{1}@{2}:{3}/{4}'
+ftpUrl = ftpString.format(user, password, server, port, filepath)
+
+
+def hello_pubsub(event, context):
+
+
+  icone_data=get_ftp_file(ftpUrl)
+  icone_obj=parse_xml(icone_data)
+  info=icone_translator.initialize_info()
+
+  wzdx_obj=icone_translator.wzdx_creator(icone_obj,info)
+  wzdx_schema=get_wzdx_schema('wzdx_schema.json')
+  validation_result=icone_translator.validate_wzdx(wzdx_obj,wzdx_schema)
+  # if not validation_result:
+  #   print('validation failed! see previous messages for error messages')
+  print(json.dumps(wzdx_obj))
+
+  publisher = pubsub_v1.PublisherClient()
+  topic_path = publisher.topic_path('cdot-cv-ode-dev', 'wzdx_messages')
```

Comment:
These variables should also come from config settings. We'll be changing them in staging and production environments and will not want to have to recompile or change code to adjust environment specific names.

Reply:
These were transitioned to environment variables.